InterGalacticCurrencyConvertor/src/main/java/DataProcessor.java

```diff
@@ -1,80 +1,73 @@
-		mStatementObjList.add(inStatement);
+	static {
+	  // Needs to be updated when new Processor classes are added
+	  mQueryProcessorList = new ArrayList<>();
+	  mQueryProcessorList.add(new InterGalacticToDecimalQueryProcessor());
+	  mQueryProcessorList.add(new MaterialCostQueryProcessor());
+	  
+	  mStatementProcessorList = new ArrayList<>();
+	  mStatementProcessorList.add(new InterGalacticToRomanMapStatementProcessor());
+	  mStatementProcessorList.add(new MaterialCostStatementProcessor());
 	}
-	public DataProcessor() {
-		loadClasses(Statement.class.getPackage().getName(), Statement.class);
-		loadClasses(Query.class.getPackage().getName(), Query.class);
+	private DataProcessor() {}
+	public static DataProcessor getInstance() {
+	  return mInstance;
 	}
-	private DataProcessResult processQuery(String [] inTokens) {
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		for (Query query: mQueryObjList) {
-			vResult = query.processQuery(inTokens);
-			if (vResult.hasSucceeded()) {
+	private DataProcessResult processQuery(ProcessorData processorData) {
+		DataProcessResult result = DataProcessResult.EMPTY_RESULT;
+		for (QueryProcessor queryProcessor: mQueryProcessorList) {
+			result = queryProcessor.processQuery(processorData);
+			if (result.isSuccess()) {
 				break;
 			}
 		}
-		return vResult;
+		return result;
 	}
-	private DataProcessResult processStatement(String [] inTokens) {
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		for (Statement stm: mStatementObjList) {
-			vResult = stm.processStatement(inTokens);
-			if (vResult.hasSucceeded()) {
+	private DataProcessResult processStatement(ProcessorData processorData) {
+		DataProcessResult result = DataProcessResult.EMPTY_RESULT;
+		for (StatementProcessor stmProcessor: mStatementProcessorList) {
+			result = stmProcessor.processStatement(processorData);
+			if (result.isSuccess()) {
 				break;
 			}
 		}
-		return vResult;
+		return result;
 	}
-	public DataProcessResult processData(String inLine) {
-	  Objects.requireNonNull(inLine, "Invalid Data input");
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		inLine = inLine.toLowerCase();
-		if (inLine.equals("exit")) {
-			DataReader.setExit(true);
-			return vResult;
+	public DataProcessResult processData(String data) {
+	  Objects.requireNonNull(data, "Invalid Data input");
+		DataProcessResult result = DataProcessResult.EMPTY_RESULT;
+		ProcessorData processorData = new ProcessorData(data);
+		if (data.trim().equalsIgnoreCase("exit")) {
+      DataReader.setExit(true);
+      return result;
 		}
-		inLine = inLine.trim();
-		inLine = inLine.replaceAll("\\s+", " ");
-		String [] vTokens = inLine.split("\\s");
-		if (vTokens == null || vTokens.length == 0) {
-			return vResult;
-		} else if (vTokens[vTokens.length-1].equals("?")) {
-			vResult = processQuery(vTokens);
+		if (data.trim().length() == 0) {
+			return result;
+		} else if (data.matches("^\\s*.+\\?\\s*$")) {
```

Comment:
This can then just be data.endsWith("?") or something similar.

Reply:
this is not done, since we have single list of processors, each processor will handle it themselves